src/Utils.cpp

```diff
@@ -364,6 +375,37 @@ static void
         power::interface, power::property);
 }
 
+static void
+    getPGoodStatus(const std::shared_ptr<sdbusplus::asio::connection>& conn,
+                   size_t retries = 2)
+{
+    conn->async_method_call(
+        [conn, retries](boost::system::error_code ec,
+                        const std::variant<int>& pgoodValue) {
+            if (ec)
+            {
+                if (retries)
+                {
+                    auto timer = std::make_shared<boost::asio::steady_timer>(
+                        conn->get_io_context());
+                    timer->expires_after(std::chrono::seconds(15));
+                    timer->async_wait(
+                        [timer, conn, retries](boost::system::error_code) {
+                            getPGoodStatus(conn, retries - 1);
+                        });
+                    return;
+                }
+
+                std::cerr << "error getting pgood status " << ec.message()
+                          << "\n";
+                return;
+            }
+            pgoodStatus = std::get<int>(pgoodValue) != 0;
```

Comment:
Am I missing how `pgoodStatus` is captured into this lambda?

Reply:
It's a global, so it doesn't have to be.

Reply:
ah... obviously havent seen globals in awhile